stm32-modules/thermocycler-refresh/firmware/CMakeLists.txt

```diff
@@ -0,0 +1,176 @@
+# This CMakeLists.txt handles everything that is compiled only when
+# cross-compiling, like the board support packages and special options.
+
+find_package(STM32G4xx)
+add_STM32G4_driver(${TARGET_MODULE_NAME})
+add_STM32G4_freertos(${TARGET_MODULE_NAME})
+add_STM32G4_usb(${TARGET_MODULE_NAME})
+
+set(SYSTEM_DIR "${CMAKE_CURRENT_SOURCE_DIR}/system")
+set(COMMS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/host_comms_task")
+
+# Add source files that should be checked by clang-tidy here
+set(${TARGET_MODULE_NAME}_FW_LINTABLE_SRCS
+  ${SYSTEM_DIR}/main.cpp 
+  #${SYSTEM_DIR}/freertos_system_task.cpp
+  ${SYSTEM_DIR}/system_policy.cpp
+  #${COMMS_DIR}/freertos_comms_task.cpp
+  ${SYSTEM_DIR}/freertos_idle_timer_task.cpp)
+
+# Add source files that should NOT be checked by clang-tidy here
+set(${TARGET_MODULE_NAME}_FW_NONLINTABLE_SRCS
+  ${SYSTEM_DIR}/system_stm32g4xx.c
+  ${SYSTEM_DIR}/stm32g4xx_it.c
+  ${SYSTEM_DIR}/startup_stm32g491vetx.s
+  ${SYSTEM_DIR}/system_hardware.c
+  #${COMMS_DIR}/usbd_conf.c
+  #${COMMS_DIR}/usbd_desc.c
+  )
+
+add_executable(${TARGET_MODULE_NAME}
+  ${${TARGET_MODULE_NAME}_FW_LINTABLE_SRCS}
+  ${${TARGET_MODULE_NAME}_FW_NONLINTABLE_SRCS})
+
+target_link_options(${TARGET_MODULE_NAME}
+  PRIVATE
+  "LINKER:-T,${SYSTEM_DIR}/STM32G491VETx_FLASH.ld"
+  "LINKER:--print-memory-usage"
+  "LINKER:--error-unresolved-symbols"
+  "LINKER:--gc-sections"
+  "LINKER:-u,_printf_float"
+  "LINKER:-u,_scanf_float")
+
+# Incurs at least a relink when you change the linker file (and a recompile of main
+# but hopefully that's quick)
+set_source_files_properties(${SYSTEM_DIR}/main.cpp
+  PROPERTIES
+  OBJECT_DEPENDS "${SYSTEM_DIR}/STM32G491VETx_FLASH.ld")
+
+set_target_properties(${TARGET_MODULE_NAME}
+  PROPERTIES CXX_STANDARD 20
+             CXX_STANDARD_REQUIRED TRUE
+             C_STANDARD 11
+             C_STANDARD_REQUIRED TRUE)
+
+  target_include_directories(${TARGET_MODULE_NAME}
+    PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include
+    )
+
+target_compile_options(${TARGET_MODULE_NAME}
+  PUBLIC
+  -Wall
+  -Werror
+  $<$<COMPILE_LANGUAGE:CXX>:-Weffc++>
+  $<$<COMPILE_LANGUAGE:CXX>:-Wreorder>
+  $<$<COMPILE_LANGUAGE:CXX>:-Wsign-promo>
+  $<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
+  $<$<COMPILE_LANGUAGE:CXX>:-Wctor-dtor-privacy>
+  $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>)
+
+
+target_link_libraries(
+  ${TARGET_MODULE_NAME}
+  STM32G4xx_Drivers_${TARGET_MODULE_NAME} 
+  STM32G4xx_FreeRTOS_${TARGET_MODULE_NAME} 
+  #STM32G4xx_USB_${TARGET_MODULE_NAME} 
```

Comment:
this is ok even though the include directories are added right

Reply:
As far as I can tell...  it's compiling/linking correctly. This'll be uncommented, I just wanted to take out the USB dependency until I've got the supporting code.